src/fmripost_aroma/workflows/resample.py

```diff
@@ -0,0 +1,189 @@
+
+    """
+    workflow = pe.Workflow(name=name)
+
+    inputnode = pe.Node(
+        niu.IdentityInterface(
+            fields=[
+                'bold_file',
+                'bold_ref_file',
+                'target_ref_file',
+                'target_mask',
+                # HMC
+                'motion_xfm',
+                # SDC
+                'fmap',
+                'fmap_id',
+                # Anatomical
+                'boldref2anat_xfm',
+                # Template
+                'anat2std_xfm',
+                # Entity for selecting target resolution
+                'resolution',
+            ],
+        ),
+        name='inputnode',
+    )
+
+    outputnode = pe.Node(
+        niu.IdentityInterface(fields=['bold_file', 'resampling_reference']),
+        name='outputnode',
+    )
+
+    gen_ref = pe.Node(GenerateSamplingReference(), name='gen_ref', mem_gb=0.3)
+
+    boldref2target = pe.Node(niu.Merge(2), name='boldref2target', run_without_submitting=True)
+    bold2target = pe.Node(niu.Merge(2), name='bold2target', run_without_submitting=True)
+    resample = pe.Node(
+        ResampleSeries(jacobian=jacobian),
+        name='resample',
+        n_procs=omp_nthreads,
+        mem_gb=mem_gb['resampled'],
+    )
+
+    workflow.connect([
+        (inputnode, gen_ref, [
+            ('bold_ref_file', 'moving_image'),
+            ('target_ref_file', 'fixed_image'),
+            ('target_mask', 'fov_mask'),
+            (('resolution', _is_native), 'keep_native'),
+        ]),
+        (inputnode, boldref2target, [
+            ('boldref2anat_xfm', 'in1'),
+            ('anat2std_xfm', 'in2'),
+        ]),
+        (inputnode, bold2target, [('motion_xfm', 'in1')]),
+        (inputnode, resample, [('bold_file', 'in_file')]),
+        (gen_ref, resample, [('out_file', 'ref_file')]),
+        (boldref2target, bold2target, [('out', 'in2')]),
+        (bold2target, resample, [('out', 'transforms')]),
+        (gen_ref, outputnode, [('out_file', 'resampling_reference')]),
+        (resample, outputnode, [('out_file', 'bold_file')]),
+    ])  # fmt:skip
+
+    if not fieldmap_id:
+        return workflow
+
+    fmap_select = pe.Node(
+        KeySelect(fields=['fmap'], key=fieldmap_id),
+        name='fmap_select',
+        run_without_submitting=True,
+    )
+    distortion_params = pe.Node(
+        DistortionParameters(metadata=metadata),
+        name='distortion_params',
+        run_without_submitting=True,
+    )
+    workflow.connect([
+        (inputnode, fmap_select, [
+            ('fmap', 'fmap_ref'),
+            ('fmap_id', 'keys'),
+        ]),
+        (inputnode, distortion_params, [('bold_file', 'in_file')]),
+        # Inject fieldmap correction into resample node
+        (fmap_select, resample, [('fmap', 'fieldmap')]),
+        (distortion_params, resample, [
+            ('readout_time', 'ro_time'),
+            ('pe_direction', 'pe_dir'),
+        ]),
+    ])  # fmt:skip
```

Comment:
Here is the section where you need to resample the fieldmap from fmap space to target space, using the inverse boldref2fmap transform, in addition to the boldref2anat and anat2MNI transforms. Then you pass in the fieldmap in MNI space to `resample`.